packages/react-reconciler/src/ReactFiberLane.js

```diff
@@ -570,17 +619,69 @@ export function markRootUpdated(root: FiberRoot, updateLane: Lane) {
 
   // Unsuspend any update at equal or lower priority.
   const higherPriorityLanes = updateLane - 1; // Turns 0b1000 into 0b0111
+  const equalOrLowerPriorityLanes = ~higherPriorityLanes;
+
+  // We use this below to mark the event time
+  const unsuspendedLanes = equalOrLowerPriorityLanes & root.suspendedLanes;
+
   root.suspendedLanes &= higherPriorityLanes;
   root.pingedLanes &= higherPriorityLanes;
+
+  if (eventTime !== -1) {
+    const eventTimes = root.eventTimes;
+    let lanes = updateLane | unsuspendedLanes;
+    while (lanes > 0) {
+      const index = ctrz(lanes);
+
+      // Assumes event times are monotonically increasing. Last one wins.
+      if (eventTimes[index] === -1) {
+        eventTimes[index] = eventTime;
+      }
+
+      lanes &= ~(1 << index);
+    }
+  }
 }
 
 export function markRootSuspended(root: FiberRoot, suspendedLanes: Lanes) {
   root.suspendedLanes |= suspendedLanes;
   root.pingedLanes &= ~suspendedLanes;
+
+  // The suspended lanes are no longer CPU-bound. Clear their event times.
+  const eventTimes = root.eventTimes;
+  let lanes = suspendedLanes;
+  while (lanes > 0) {
+    const index = ctrz(lanes);
+
+    eventTimes[index] = -1;
```

Comment:
Curious why we use an explicit constant for e.g. `NoLane = 0` but we pass around a naked `-1` in so many places. Should we use a constant for this too (to make it easier to grep for later if nothing else)?

Reply:
Yeah I can do that. It won't add any type safety, unfortunately, but it does make it more obvious.